Move to GitHub Actions (https://github.com/KWB-R/kwb.pkgbuild/issues/78)

mrustl opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kwb-r/kwb.prep/3)
<!-- Reviewable:end -->

mrustl committed:
Move to GitHub Actions (https://github.com/KWB-R/kwb.pkgbuild/issues/78)

mrustl committed:
Add "gh-actions" folder to Rbuildignore

hsonne committed:
Look in package dir for config file, closes #4

hsonne committed:
Look for global variable "user_config_dir"

hsonne committed:
Use user's config dir to lookup types

hsonne committed:
Look in user's config folder

hsonne committed:
By default, now look in user's config folder

mrustl committed:
Merge pull request #5 from KWB-R/fix-4

Look in package dir for config file, closes #4

hsonne reviewed (approved):
I trust you to have copied this from some trustable source

mrustl commented:
Yes.

Copied all first from: 
https://github.com/KWB-R/pkgmeta/tree/570102d5c97abae7246f6fee2fe896b0047aba74/.github/workflows
(which I once copied and modfied from one "trustworthy" resource), 

Afterwords replaced "pkgdown" worklow with https://github.com/r-lib/actions/blob/92c6da7b47f0c1814c4d62d62a9c0f24291fe5de/.github/workflows/pkgdown.yaml and the rest from

mrustl commented:
@hsonne let`s wait with merge. Will add the missing platforms (linux, mac-os) and R versions (as discussed in. https://github.com/KWB-R/kwb.pkgbuild/issues/78#issuecomment-730975905)

hsonne committed:
Add find_identical_columns()

hsonne committed:
Improve find_identical_columns()

- Use helper function record()
- Make sure that result is a data frame with
  columns "group", "column_a", "column_b"
- Add general utility function names_which()

hsonne committed:
Merge branch 'dev' into github-actions

hsonne committed:
Change a, b to x, y, keep "group" column right

Merge branch 'dev' into github-actions

# Conflicts:
#	R/find_identical_columns.R

hsonne committed:
Merge branch 'dev' into github-actions

hsonne committed:
Merge branch 'dev' into github-actions

hsonne committed:
Merge branch 'dev' into github-actions

mrustl committed:
CI: add multi-plattform and r-versions (https://github.com/KWB-R/kwb.pkgbuild/issues/78)

os: linux, mac, win (always r-version "release")
r-versions: "oldrel", "release" and "devel" (only on windows all 3 are tested)

Code inspired by:
https://github.com/r-lib/actions/blob/a22e2286d053b722d515f5cc394c41231b1316cb/examples/check-full.yaml#L34

https://github.com/r-lib/actions/blob/ada5c384043d8d96dfbfd8fc45940fdee3c155a6/.github/workflows/check-standard.yaml

mrustl commented:
@hsonne your package runs nicely on all plattforms and R versions. So it`s ready for "dev" branch

hsonne commented:
> 
> 
> @hsonne your package runs nicely on all plattforms and R versions. So it`s ready for "dev" branch

This is great. Thanks for your efforts. I accidentally commited changes to this branch which were meant to go to dev. I will merge this very soon.

hsonne committed:
Replace "UrbanRunoff" with "kwb.prep" in README

hsonne committed:
Replace "urbanAnnualRunoff" with "kwb.prep"

hsonne reviewed (approved):
I changed "urbanAnnualRunoff" with "kwb.prep" and hope that this works

mrustl commented:
> 
> 
> I changed "urbanAnnualRunoff" with "kwb.prep" and hope that this works

Opps....that`s why this needs to go asap to https://github.com/KWB-R/kwb.pkgbuild/issues/78
